haystack/components/embedders/openai_text_embedder.py

```diff
@@ -139,6 +147,27 @@ def from_dict(cls, data: Dict[str, Any]) -> "OpenAITextEmbedder":
         deserialize_secrets_inplace(data["init_parameters"], keys=["api_key"])
         return default_from_dict(cls, data)
 
+    def _prepare_input(self, text: str) -> Dict[str, Any]:
+        if not isinstance(text, str):
+            raise TypeError(
+                "OpenAITextEmbedder expects a string as an input."
+                "In case you want to embed a list of Documents, please use the OpenAIDocumentEmbedder."
+            )
+
+        text_to_embed = self.prefix + text + self.suffix
+
+        # copied from OpenAI embedding_utils (https://github.com/openai/openai-python/blob/main/openai/embeddings_utils.py)
+        # replace newlines, which can negatively affect performance.
+        text_to_embed = text_to_embed.replace("\n", " ")
+
+        kwargs = {"model": self.model, "input": text}
```

Comment:
It's would be better to do `kwargs: Dict[str, Any] = ...` here to avoid type issues